tests/test_sync.py

```diff
@@ -153,11 +161,55 @@ def test_push_template_branch_error(self):
         open(test_fn, "a").close()
         psync.commit_template_changes()
         # Try to push changes
-        try:
+        with pytest.raises(nf_core.sync.PullRequestException) as exc_info:
             psync.push_template_branch()
-            raise UserWarning("Should have hit an exception")
-        except nf_core.sync.PullRequestException as e:
-            assert e.args[0].startswith("Could not push TEMPLATE branch")
+        assert exc_info.value.args[0].startswith("Could not push TEMPLATE branch")
+
+    def test_create_merge_base_branch(self):
+        """Try creating a merge base branch"""
+        psync = nf_core.sync.PipelineSync(self.pipeline_dir)
+        psync.inspect_sync_dir()
+        psync.get_wf_config()
+
+        psync.create_merge_base_branch()
+
+        assert psync.merge_branch in psync.repo.branches
+
+    def test_create_merge_base_branch_thrice(self):
```

Comment:
Just curious, why would it work once, but not three times?

Reply:
The first time you call this function, the merge branch does not exist yet (it is only created at the end of this function), so you don't go into the `if`-statement at L274. Also, the second time you call this function, the existing merge branch only has the base format, i.e. without the `-{branch_no}` at the end, so you need to call it a third time to make sure this is picked up.

https://github.com/Aratz/nf-core-tools/blob/dev/nf_core/sync.py#L274-L286

Alternatively, we could set the merge branch format to include the branch number suffix from the beginning, so you only need to test it twice to cover all the code.

Reply:
makes totally sense now. thanks! Do you mind putting this in as a comment, so we don't forget about the logic. also there is a merge conflict now in the changelog. but than we are good to go!

Reply:
Yes will do! :+1:

Reply:
Should be all good now